woudc_data_registry/epicentre/instrument.py

```diff
@@ -188,13 +187,26 @@ def update(ctx, identifier, station, dataset,
     """Update instrument information"""
 
     instrument_ = {}
+    registry_ = registry.Registry()
```

Comment:
Notice that other parts of the file are using functions from `epicenter/metadata` to search, insert, and update to the registry instead of accessing the registry directly (and re-inventing the wheel). I recommend using the `get_metadata` function from there, look around to see how other modules use it.